app/models/acts_as_constrained/concerns/constrained.rb

```diff
@@ -56,12 +62,35 @@ def self.constrain_by constraint_type, options = {}
       #
       # MyConstrainedModel.constrained_by_date(Date.tomorrow).
       #   constrained_by_dummy(99)
+      #
+      # Some constrains may be optional. In that case it will try to get results with them (or fallback to the
+      # the required ones). The concatenation of more scopes will not work in this scenario.
       scope :constrained_by, ->(constraints) do
-        constraints.inject(self) { |result, constraint|
+        req_constraints = constraints.reject{|c| (self.optional_constraints || []).include?(c)}
+        opt_constraints = constraints.select{|c| (self.optional_constraints || []).include?(c) && constraints[c].present?}
+
+        #We build the query with the required constraints and then we try to get results with the optional ones
+        result = result_with_required = build_constraints self, req_constraints
+        loop do
```

Comment:
What does this loop mean? If no results are found the last optional constraint is removed and then try again? Should we create some additional option that stores the priority so we remove the least important one? Or the order of insertion is a good enough criteria? If the order is good enough we should document it in the README

In any case, it would be great to comment this loop so future programmers understand why it's important.